### PR TITLE
Updates developer guide and removes integtest.sh

### PR DESCRIPTION
**Pull request by DarshitChanpura**
### Description
Updates developer guide to reflect changes in demo config installation tool and removes integtest.sh as it is no longer used: https://github.com/opensearch-project/opensearch-build/issues/497

### Issues Resolved
- Resolves https://github.com/opensearch-project/security/issues/3694

### Check List
- [ ] ~New functionality includes testing~
- [x] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by derek-ho — APPROVED** [review]
LGTM

**Review by willyborankin — DISMISSED** [review]

File: DEVELOPER_GUIDE.md [hunk 1]
```diff
@@ -162,6 +162,21 @@ extension_hw_greet:
     - "hw-user"
 ```
 
+### Setting up password for demo admin user
+
+This step is a pre-requisite to installing demo configuration. There are two ways password can be set for demo `admin` user:
```
Comment (scrawfor99) [inline DEVELOPER_GUIDE.md:167]:
```suggestion
This step is a pre-requisite to installing demo configuration. There are two ways the demo `admin` user password can be set:
```

File: DEVELOPER_GUIDE.md [hunk 2]
```diff
@@ -162,6 +162,21 @@ extension_hw_greet:
...
+1. **Setting via environment variable:** Password be passed by exporting `OPENSEARCH_INITIAL_ADMIN_PASSWORD` variable with a password.
```
Comment (scrawfor99) [inline DEVELOPER_GUIDE.md:168]:
```suggestion
1. **Setting via environment variable:** The default password  can be passed by exporting `OPENSEARCH_INITIAL_ADMIN_PASSWORD` variable with a password.
```

File: DEVELOPER_GUIDE.md [hunk 3]
```diff
@@ -162,6 +162,21 @@ extension_hw_greet:
...
+```shell
+export OPENSEARCH_INITIAL_ADMIN_PASSWORD=<password>
+```
+
+2. **Setting via a text file:** Password can also be supplied by creating a single line text file `opensearch_initial_admin_password.txt` file with `<password>` under `$OPENSEARCH_HOME/config/` folder.
```
Comment (scrawfor99) [inline DEVELOPER_GUIDE.md:173]:
```suggestion
2. **Setting via a text file:** The password can also be supplied by creating a single line text file `opensearch_initial_admin_password.txt` file with `<password>` under `$OPENSEARCH_HOME/config/` folder.
```

File: DEVELOPER_GUIDE.md [hunk 4]
```diff
@@ -162,6 +162,21 @@ extension_hw_greet:
...
+
+**_Note:_** If no password is supplied the installation will fail. The passwords supplied need to be strong as there is a password validation check in place. There is an option to skip password validation by passing `-t` option to installation script. However, this should only be used for test environments.
```
Comment (scrawfor99) [inline DEVELOPER_GUIDE.md:175]:
```suggestion
**_Note:_** If no password is supplied, the installation will fail. The password supplied will also be tested for its strength and will be blocked if it is too simple. There is an option to skip this password validation by passing the `-t` option to the installation script. However, this should only be used for test environments.
```

**Review by scrawfor99 — COMMENTED** [review]
Thanks DC; I left some suggestions on wording. I am not sure if you are intending to use "password" as a proper noun or not. I assume no, in which case I would suggest taking a peak at my suggestions. If you do mean it to be a proper noun, you are missing capitalization in a couple spots. (First comment and then later)

**Comment by DarshitChanpura** [comment]
@scrawfor99 Addressed all feedback. @willyborankin would you please re-review this?

**Review by willyborankin — DISMISSED** [review]

**Review by willyborankin — APPROVED** [review]

**Comment by DarshitChanpura** [comment]
CI failures address via https://github.com/opensearch-project/security/pull/3850

File: DEVELOPER_GUIDE.md (hunk 3 above)
Comment (peternied) [reply DEVELOPER_GUIDE.md]: @DarshitChanpura Didn't reading password from a file get removed, seems like this should also be updated.

**Review by peternied — CHANGES_REQUESTED** [review]
CI should pass on this readme only change, no?

File: DEVELOPER_GUIDE.md (hunk 3 above)
Comment (DarshitChanpura) [reply DEVELOPER_GUIDE.md]: Yes, it has been removed

**Comment by DarshitChanpura** [comment]
@scrawfor99 @peternied mind giving this PR another lookeru?

**Review by peternied — APPROVED** [review]

**Pull request merged**